pkg/operator/sync.go

```diff
@@ -39,6 +39,10 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 		glog.V(3).Info("Synced up all machine-api-controller components")
 	}
 
+	if config.Controllers.TerminationHandler != clusterAPIControllerNoOp {
```

Comment:
should we do this within `syncClusterAPIController` so it's covered by all the `status*` logic?

Reply:
Which status logic are you referring to here? Do you mean the waiting for the rollouts? I don't think there's any duplication of `status` apart from the waiting, but they have to be different because the fields are different in DaemonSets vs Deployments

As far as I can tell the status is only updated once this method returns